python-backend/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -0,0 +1,53 @@
+import uuid
+
+from flask_restplus import Resource, reqparse
+from flask import request
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_admin
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.minespace_user import MinespaceUser
+from ..models.minespace_user_mine import MinespaceUserMine
+from app.extensions import db
+
+
+class MinespaceUserResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('email', type=str, location='json', required=True)
+    parser.add_argument('mine_guids', type=list, location='json', required=True)
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def get(self, user_id=None):
+        if user_id or request.args.get('email'):
+            user = MinespaceUser.find_by_id(user_id)
+            if not user:
+                user = MinespaceUser.find_by_email(request.args.get('email'))
+                if not user:
+                    return self.create_error_payload(404, "user not found"), 404
+            return user.json()
+        users = MinespaceUser.query.filter_by(deleted_ind=False).all()
+        return {'users': [x.json() for x in users]}
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def post(self, user_id=None):
+        if user_id:
+            return self.create_error_payload(400, "unexpected user guid"), 400
+        data = self.parser.parse_args()
+        new_user = MinespaceUser(email=data.get('email'))
```

Comment:
I'm seeing a mix of using class constructors like this as well as a `create_classname` class method throughout the codebase. Last I spoke to @FreshGyan (while building mine types), I believe the consensus was to use the class method approach.

As a team, I think we should determine which approach we're going to use.

Perhaps this was discussed in the "fatter" models part of the db standards presentation and I have since forgotten, but the following lines of code (db interfacing + record management) seem better suited for a model-level method to me.

Reply:
Yea, old habit, having create methods is the pattern we are going to go with going forward. Refactored

Reply:
Old habit, refactored to use create methods.